imports/plugins/included/product-variant/containers/productsContainerCustomer.js

```diff
@@ -3,118 +3,131 @@ import PropTypes from "prop-types";
+      const { isLoading } = this.state;
+
+      const { pageInfo = {} } = catalogItems;
+      const { hasNextPage } = pageInfo;
+      const products = (catalogItems.edges || []).map((edge) => edge.node.product);
+
       return (
         <Comp
           {...this.props}
-          loadProducts={this.loadProducts}
+          showNotFound={shouldShowNotFound}
+          canLoadMoreProducts={hasNextPage}
+          isLoading={isLoading}
+          loadProducts={this.handleLoadProducts}
+          products={products}
+          shopCurrencyCode={Reaction.getPrimaryShopCurrency()}
         />
       );
     }
   }
 );
 
+/**
+ * @name composer
+ * @private
+ * @summary Sets up tagSlugOrId prop that is passed to withTag GraphQL HOC,
+ *  unless tag not found, in which case sets skips GraphQL HOCs and shows not found view
+ * @param {Object} props - Props passed down from parent components
+ * @param {Function} onData - Callback to execute with props
+ * @returns {undefined}
+ */
 function composer(props, onData) {
   window.prerenderReady = false;
 
-  let canLoadMoreProducts = false;
-
-  const queryParams = {};
-  const slug = Reaction.Router.getParam("slug");
-  const shopIdOrSlug = Reaction.Router.getParam("shopSlug");
-  let tagIdForPosition = "_default";
-
-  if (slug) {
-    const tag = Tags.findOne({ slug }) || Tags.findOne({ _id: slug });
+  if (!Meteor.userId()) {
+    return;
+  }
 
-    // if we get an invalid slug, don't return all products
+  const tagSlugOrId = Reaction.Router.getParam("slug");
+  if (tagSlugOrId) {
+    const tag = Tags.findOne({ slug: tagSlugOrId }) || Tags.findOne({ _id: tagSlugOrId });
     if (!tag) {
       onData(null, {
-        showNotFound: true
+        shouldShowNotFound: true,
+        // Skip loading GraphQL data via HOCs (withPrimaryShopId, withTag, withCatalogItems, etc),
+        // since we are going to render the 404 view
+        shouldSkipGraphql: true
       });
-
       return;
     }
-    queryParams.tagIds = [tag._id];
-    tagIdForPosition = tag._id;
-  }
-
-  if (shopIdOrSlug) {
-    queryParams.shopIdsOrSlugs = [shopIdOrSlug];
-  }
-
-  const queryString = Reaction.Router.current().query;
-  if (queryString) {
-    queryParams.query = queryString.query;
   }
 
-  const scrollLimit = Session.get("productScrollLimit");
-  const productsSubscription = Meteor.subscribe("Products/grid", scrollLimit, queryParams);
+  // TODO multi-shop support - const shopIdOrSlug = Reaction.Router.getParam("shopSlug");
+  // Currently no way to get a shop's graphql ID by mongo _id or slug
 
-  if (productsSubscription.ready()) {
-    window.prerenderReady = true;
-  }
-
-  const catalogCursor = Catalog.find({
-    "product.type": "product-simple"
-  }, {
-    $sort: {
-      [`product.positions.${tagIdForPosition}.position`]: 1,
-      createdAt: -1
-    }
-  });
+  // TODO graphql custom tag sort - $sort: { [`product.positions.${tagIdForPosition}.position`]: 1 }
```

Comment:
We no longer support custom positioning so everything using `product.positions` can just be removed.